terraform/layer2/main.tf

```diff
@@ -14,17 +14,17 @@ terraform {
 }
 
 provider "aws" {
-  region = data.terraform_remote_state.layer1.outputs.region
+  region = var.region
 }
 
 
 locals {
   # layer 1 remote state refs
-  region     = var.region
-  bucket     = var.bucket
-  bucket_key = "${var.bucket_key}-layer1"
-  dynamodb   = var.dynamodb
+  bucket_key_layer1 = "${var.bucket_key}-layer1"
```

Comment:
is the name `bucket_key_layer1`  the same use for storing the state in level1?

Reply:
Yeah - I was starting to get confused between bucket key for current layer and bucket key for the remote state... so I explicitly name id `bucket_key_layer1`

Reply:
The bucket referenced in layer2 must be the same as the key it was assigned when storing the state in level1

Reply:
Yep - I ran into that issue! 😅 But it works now / deploys okay because each layer's bucket key is deterministically named  based on the layer number. I need derived names so that my makefile works regardless of the layer and regardless of whether it's doing an init or not.